src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -194,6 +198,29 @@ JoinTreeContainsSubquery(Query *query)
 }
 
 
+bool
+JoinTreeContainsWithoutFrom(Query *query)
+{
+	if (query->rtable == NIL)
+	{
+		return true;
+	}
+
+#if PG_VERSION_NUM >= 120000
+	else if (list_length(query->rtable) == 1)
```

Comment:
Hmm, I think I misguided you on this. It looks like for simple  `INSERT INTO tbl VALUES ();` type of queries would also end up with `RTE_RESULT ` appended next to RTE_RELATION.  I want to test/play with that to see if anything is broken on Citus with such a change. 

But as I mentioned I'm blocked on `CREATE EXTENSION citus` is failing on my local machine.

For reference, the query tree of an INSERT after RTE_RESULT changes:

```
   {QUERY 
   :commandType 3 
   :querySource 0 
   :canSetTag true 
   :utilityStmt <> 
   :resultRelation 1 
   :hasAggs false 
   :hasWindowFuncs false 
   :hasTargetSRFs false 
   :hasSubLinks false 
   :hasDistinctOn false 
   :hasRecursive false 
   :hasModifyingCTE false 
   :hasForUpdate false 
   :hasRowSecurity false 
   :cteList <> 
   :rtable (
      {RTE 
      :alias <> 
      :eref 
         {ALIAS 
         :aliasname users_table 
         :colnames ("user_id" "time" "value_1" "value_2" "value_3" "value_4")
         }
      :rtekind 0 
      :relid 16393 
      :relkind r 
      :rellockmode 3 
      :tablesample <> 
      :lateral false 
      :inh false 
      :inFromCl false 
      :requiredPerms 1 
      :checkAsUser 0 
      :selectedCols (b)
      :insertedCols (b 8)
      :updatedCols (b)
      :extraUpdatedCols (b)
      :securityQuals <>
      }
      {RTE 
      :alias <> 
      :eref 
         {ALIAS 
         :aliasname *RESULT* 
         :colnames <>
         }
      :rtekind 8 
      :lateral false 
      :inh false 
      :inFromCl false 
      :requiredPerms 0 
      :checkAsUser 0 
      :selectedCols (b)
      :insertedCols (b)
      :updatedCols (b)
      :extraUpdatedCols (b)
      :securityQuals <>
      }
   )
   :jointree 
      {FROMEXPR 
      :fromlist (
         {RANGETBLREF 
         :rtindex 2
         }
      )
      :quals <>
      }
   :targetList (
      {TARGETENTRY 
      :expr 
         {CONST 
         :consttype 23 
         :consttypmod -1 
         :constcollid 0 
         :constlen 4 
         :constbyval true 
         :constisnull false 
         :location 42 
         :constvalue 4 [ 8 0 0 0 0 0 0 0 ]
         }
      :resno 1 
      :resname user_id 
      :ressortgroupref 0 
      :resorigtbl 0 
      :resorigcol 0 
      :resjunk false
      }
   )
   :override 0 
   :onConflict <> 
   :returningList <> 
   :groupClause <> 
   :groupingSets <> 
   :havingQual <> 
   :windowClause <> 
   :distinctClause <> 
   :sortClause <> 
   :limitOffset <> 
   :limitCount <> 
   :rowMarks <> 
   :setOperations <> 
   :constraintDeps <> 
   :withCheckOptions <> 
   :stmt_location 0 
   :stmt_len 44
   }
```

Reply:
Checking for explicit `length == 1` seems all the more correct then. It's more that we want the code to be for-all rather than for-any, which my earlier code was the latter. The length check will give the incorrect result if we end up with rtable having length greater than 1 & containing only `RTE_RESULT`